File: BankAccount.java
public class BankAccount {

    protected String name;
    protected double balance;

    public BankAccount(String n, double b){
        name = n;
        balance = b;
    }

    public String getName(){
        return name;
    }

    public double getBalance(){
        return balance;
    }

    //a method to add to the bankAccount
    public double deposit(double depNum){
        balance += depNum;
        CheckingAccount.transactionCount += 1;
        return balance;
    }

    //a method to take out of the bankAccount
    public double withdraw(double wdNum){
        balance -= wdNum;
        CheckingAccount.transactionCount += 1;
        return balance;
    }

    //a method to make a transfer from one account to another
    public void transfer(BankAccount originAccount, double tranNum){
        originAccount.balance += tranNum;
        this.balance -= tranNum;
    }

    public String toString(){
        return name + ": $" + balance;
    }

}

File: BankAccount.cpp
//
//  BankAccount.cpp
//  BankAccount
//
//  Created by Rebekah on 11/20/18.
//  Copyright © 2018 Rebekah. All rights reserved.
//
//#include <cstring>
#include "BankAccount.hpp"
#include <string>
#include <iostream>

using namespace std;

BankAccount :: BankAccount(){
    this -> name = "";
    this -> balance = 0.0;
}

BankAccount :: BankAccount(string n, double b){
    this -> name = n;
    this -> balance = b;
    
}

string BankAccount :: getName(){                      //returns the name of a BankAccount
    return name;
}

double BankAccount:: getBalance(){                    //returns the balance of a BankAccount
    return balance;
}

string BankAccount::toString(){                       //returns the name and balance of a BanckAccount
    std:: string newDoub = std:: to_string(this -> getBalance());
    string retStr = "The balance of " + this->getName() + " is " + newDoub + ".";
    return retStr;
}

/*
 a method to deposite money into a bank account
 */
void BankAccount::deposit(double a){
    this -> balance = this -> balance + a;
}

/*
 a method to withdraw money into a bank account
 */
void BankAccount::withdraw(double a){
    this -> balance = this -> balance - a;
}
